COMET.Web.Common/ViewModels/Components/LoginViewModel.cs

```diff
@@ -117,11 +144,42 @@ public async Task ExecuteLogin()
         }
 
         /// <summary>
-        /// Resets the <see cref="AuthenticationDto"/> property to a new object with the default parameters
+        /// Attempt to login to a COMET Server
+        /// </summary>
+        /// <param name="authenticationSchemeKind">The <see cref="AuthenticationSchemeKind" /> that should be used</param>
+        /// <param name="authenticationInformation">The <see cref="AuthenticationInformation" /> that should be used for authentication</param>
+        /// <returns>A <see cref="Task" /></returns>
+        public async Task ExecuteLogin(AuthenticationSchemeKind authenticationSchemeKind, AuthenticationInformation authenticationInformation)
+        {
+            this.IsLoading = true;
+            this.AuthenticationResult = await this.authenticationService.Login(authenticationSchemeKind, authenticationInformation);
+            this.IsLoading = false;
+        }
+
+        /// <summary>
+        /// Request supported <see cref="AuthenticationSchemeKind" /> by the server that we want to reach
+        /// </summary>
+        /// <returns>An awaitable <see cref="Task" /></returns>
+        public async Task RequestAvailableAuthenticationScheme()
+        {
+            this.IsLoading = true;
+
+            if (!string.IsNullOrEmpty(this.ServerConnectionService.ServerConfiguration.ServerAddress))
+            {
+                this.AuthenticationDto.SourceAddress = this.ServerConnectionService.ServerConfiguration.ServerAddress;
+            }
+
+            this.AuthenticationSchemeResponseResult = await this.authenticationService.RequestAvailableAuthenticationScheme(this.AuthenticationDto.SourceAddress, this.AuthenticationDto.FullTrust);
```

Comment:
curious to know how the `AuthenticationSchemeRespons` is being serialized. Is it the integer representation or do we serialize the values as strings, i.e. `Basic`, `LocalJwtBearer`, `ExternalJwtBearer`. My preference would be the latter, otherwise a change in order of the enums in the source file might cause inconsistencies when the SDK's on server/client side are not of the same version.

Reply:
They are send as string from the server